### DISCUSSION TITLE
How to show call hierarchy?

### DISCUSSION THREAD

**Discussion by sp71** [LazyVim/LazyVim#2279] [Q&A]
I use Golang with lazyvim but unsure how to show the call hierarchy visually? I believe the LSP already supports this but unsure how to visually bring it up

**Comment by dpetka2001** [comment]
`:lua vim.lsp.buf.outgoing_calls()` (and `incoming_calls` respectively) is what you're looking for (you could also just map the lua callbacks to your keymaps of choosing). I think [lspsaga](https://github.com/nvimdev/lspsaga.nvim) also provides this [feature](https://nvimdev.github.io/lspsaga/callhierarchy/).

**Reply by sp71** [reply]
When you call the commands directly, it only shows a single level. Unable to drill down the full call stack. Seems the visual portion is not implemented natively. 

Created feature request in [neoTree](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/1277) as lazyvim already supports NeoTree

**Reply by jyuan0** [reply]
There's also `:Telescope lsp_incoming_calls` and `lsp_outgoing_calls` (Telescope pickers).

**Reply by sp71** [reply]
Yeah same restriction with telescope options. Cool to know this exists tho
